### PR TITLE
Take in G.pkl and output input files needed for STransE

### PR DESCRIPTION
**Pull request by villanuevab**
Closes #91

### CODE REVIEW

File: relation_prediction/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    edges = G.edges()
+    print('{} edges in G'.format(len(edges)))
+    for e in edges:
+        for k, v in G.get_edge_data(*e).items():
+            relations.add(v['relation_id'])
+    relation2id = dict()
+    rid = 0
+    for r in relations:
+        relation2id[r] = rid
+        rid += 1
```
Comment (marshuang80) [inline relation_prediction/preprocessing.py:20]: This can be done in one loop but not a big deal at all

File: relation_prediction/preprocessing.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+
+def _gen_entity2id(nodes, outdir):
+    outpath = os.path.join(outdir, 'entity2id.txt')
+    print('{} nodes in G'.format(len(nodes)))
+    entity2id = dict()
+    eid = 0
+    for e in nodes:
+        entity2id[e] = eid
+        eid += 1
```
Comment (marshuang80) [inline relation_prediction/preprocessing.py:37]: Maybe using `enumerate` would be more pythonic, but again this is nitpicking

File: relation_prediction/preprocessing.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    testpath = os.path.join(outdir, 'test.txt')
+
+    examples = list()
+    for e in G.edges():
+        for k, v in G.get_edge_data(*e).items():
+            ex = '{}\t{}\t{}\n'.format(e[0], v['relation_id'], e[1])
+            examples.append(ex)
+
+    random.shuffle(examples)
+    valprop = (1.-trainprop)/2.
```
Comment (marshuang80) [inline relation_prediction/preprocessing.py:67]: I am more used to seeing the validation set to have the same size of the test set instead of training set. Not a big deal tho

Comment (villanuevab) [reply relation_prediction/preprocessing.py:67]: `valprop` is the same as `testprop` here. For example, if trainprop is 0.8, then valprop = (1. - .8)/2 = 0.1. The remaining 0.1 is used for testprop.

Comment (marshuang80) [reply relation_prediction/preprocessing.py:67]: Ah my mistake
